src/CommitmentsV2/SFA.DAS.CommitmentsV2.MessageHandlers/EventHandlers/CohortWithChangeOfPartyCreatedEventHandlerForEmail.cs

```diff
@@ -50,7 +64,33 @@ public async Task Handle(CohortWithChangeOfPartyCreatedEvent message, IMessageHa
                 tokens));
 
             _logger.LogInformation($"Sent SendEmailToEmployerCommand with template: {templateName}");
+        }
+
+        private async Task SendProviderEmail(CohortWithChangeOfPartyCreatedEvent message, IMessageHandlerContext context, GetCohortSummaryQueryResult cohortSummary)
+        {
+            var apprenticeNamePossessive = await GetApprenticeNamePossessive(message.ApprenticeshipId);
+
+            var tokens = new Dictionary<string, string>
+                {
+                    { "Subject", $"{cohortSummary.LegalEntityName} has requested that you add details on their behalf" },
+                    { "TrainingProviderName" , cohortSummary.ProviderName},
+                    { "EmployerName" , cohortSummary.LegalEntityName},
+                    { "ApprenticeNamePossessive" , apprenticeNamePossessive },
+                    { "RequestUrl", $"{cohortSummary.ProviderId}/apprentices/{cohortSummary.CohortReference}/details" }
+                };
+
+            await context.Send(new SendEmailToProviderCommand(cohortSummary.ProviderId.Value, "ProviderApprenticeshipChangeOfProviderRequested", tokens));
+
+            _logger.LogInformation($"Sent SendEmailToProviderCommand with template: ProviderApprenticeshipChangeOfProviderRequested");
+        }
+
+        internal async Task<string> GetApprenticeNamePossessive(long apprenticeshipId)
+        {
+            var apprenticeship = await _mediator.Send(new GetApprenticeshipQuery(apprenticeshipId));
```

Comment:
I think this call to GetApprenticehsipQuery is not going to work from MessageHandler, have you tried it?